ui/components/ui/hardware-wallet-state/hardware-wallet-state.js

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+  getHardwareWalletDevice,
+  getHardwareWalletPath,
+} from '../../../selectors';
+import { isDeviceAccessible } from '../../../store/actions';
+import { I18nContext } from '../../../contexts/i18n';
+import { HARDWARE_CHECK_RATE } from '../../../../shared/constants/hardware-wallets';
+import { SEVERITIES } from '../../../helpers/constants/design-system';
+import { BannerAlert } from '../../component-library';
+
+const noOp = () => {
+  // do nothing
+};
+
+/**
+ * Component that monitors hardware wallet state and displays a warning if locked
+ *
+ * @param options0
+ * @param options0.pollingRateMs
+ * @param options0.initialStatus
+ * @param options0.onUpdate
+ */
+export default function HardwareWalletState({
+  pollingRateMs = HARDWARE_CHECK_RATE,
+  initialStatus = 'locked',
+  onUpdate = noOp,
+}) {
+  const t = useContext(I18nContext);
+  const [status, setStatus] = useState(initialStatus);
+  const device = useSelector(getHardwareWalletDevice);
+  const path = useSelector(getHardwareWalletPath);
+
+  const updateHardwareLockState = useCallback(async () => {
+    const unlocked = await isDeviceAccessible(device, path);
+    const state = unlocked ? 'unlocked' : 'locked';
+    setStatus(state);
+    onUpdate(state);
+  }, [device, path, onUpdate]);
+
+  useEffect(() => {
+    const intervalId = setInterval(updateHardwareLockState, pollingRateMs);
+    return () => clearInterval(intervalId);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pollingRateMs]);
+
+  return (
+    status === 'locked' && (
+      <BannerAlert severity={SEVERITIES.DANGER}>
+        {t('ledgerLocked')}
+      </BannerAlert>
```

Comment:
I was going to suggest separating the description into a heading and description but fitting it all into the description at a smaller font size works too. Checked storybook build looks good!

<img width="947" alt="Screenshot 2023-02-21 at 10 24 26 PM" src="https://user-images.githubusercontent.com/8112138/220530612-f2e81ead-a652-4f4c-b37d-6bc7c63000fe.png">

Reply:
yeah, I just based it off the `create-account/connect-hardware` page. The message is relatively long, so already takes up some space in the popup modal - though happy to add a title if you recommend it (just let me know which message to use)

Reply:
Fine like this I think, have you got a design contact for this ticket?

Reply:
ok, thanks. Do you mean a contact in the design team? I haven't been given anyone's details.